Libplanet/Net/PreloadState.cs

```diff
@@ -0,0 +1,50 @@
+namespace Libplanet.Net
+{
+    [Equals]
+    public class PreloadState
+    {
+        protected PreloadState(PreloadPhase currentPhase, int totalPhase)
+        {
+            CurrentPhase = (int)currentPhase;
+            TotalPhase = totalPhase;
+        }
+
+        /// <summary>
+        /// Enum to describe phases of progress in preload.
+        /// </summary>
+        protected enum PreloadPhase
```

Comment:
By the way, IMHO this enum is not necessary, and quite redundant as the same hierarchy appears in the classes and this enum.